swagger/operation.go

```diff
@@ -171,6 +195,80 @@ func PagingParam(op *spec.Operation) (spec.Parameter, bool) {
 	return spec.Parameter{}, false
 }
 
+// PagingResourcePath returns the path to the array to page over for a
+// paging-enabled endpoint (specified by x-paging.resourcePath). Panics if
+// paging is not enabled.
+func PagingResourcePath(op *spec.Operation) []string {
+	pagingConfig, ok := op.Extensions["x-paging"].(map[string]interface{})
+	if !ok {
+		panic(fmt.Errorf("no paging for operation %s", op.ID))
+	}
+	resourcePath, ok := pagingConfig["resourcePath"].(string)
+	if !ok {
+		return []string{}
+	}
+	return strings.Split(resourcePath, ".")
+}
+
+// resolveSchema dereferences the $ref in this schema, if there is one
+func resolveSchema(s *spec.Swagger, schema *spec.Schema) *spec.Schema {
+	if schema.Ref.String() == "" {
+		return schema
+	}
+	refObj, _, err := schema.Ref.GetPointer().Get(s)
+	if err != nil {
+		panic(fmt.Errorf("bad schema reference %s", schema.Ref.String()))
+	}
+	r, ok := refObj.(spec.Schema)
+	if !ok {
+		panic(fmt.Errorf("bad schema reference %s", schema.Ref.String()))
+	}
+	return &r
+}
+
+// PagingResourceType returns the type of the items of the array set to page
+// over for this operation and whether the array should consist of pointers to
+// that type. Panics if paging is not enabled.
+func PagingResourceType(s *spec.Swagger, op *spec.Operation) (string, bool, error) {
+	var schema *spec.Schema
+	for statusCode := range op.Responses.StatusCodeResponses {
+		if statusCode < 400 {
+			schema = outputSchema(s, op, statusCode)
+			break
+		}
+	}
+	if schema == nil {
+		return "", false, fmt.Errorf("operation has no success type")
+	}
+
+	path := PagingResourcePath(op)
+	for idx, pathComponent := range path {
+		schema = resolveSchema(s, schema)
+		nextSchema, ok := schema.Properties[pathComponent]
+		if !ok {
+			return "", false, fmt.Errorf("could not resolve x-paging.resourcePath: "+
```

Comment:
The validation code actually uses this function to catch errors... it seemed easier than exposing the `resolveSchema` function and re-implementing the logic here.